libtrap/src/trap.c

```diff
@@ -2185,12 +2185,67 @@ static inline int trapifc_out_construct(trap_ctx_priv_t *ctx, trap_ifc_spec_t *i
    return EXIT_FAILURE;
 }
 
+trap_ctx_t *trap_ctx_init3(const char *name, const char *description, int8_t i_ifcs, int8_t o_ifcs, const char *ifc_spec, const char *service_ifc_name)
+{
+   trap_ctx_t *res = NULL;
+   trap_module_info_t module_info;
+   trap_ifc_spec_t ifcs;
+   int argc = 2;
+
+   char *argv[2] = {"-i", (char *) ifc_spec};
+
+   /* Prepare module info */
+   if (name != NULL) {
+      module_info.name = strdup(name);
+   } else {
+      module_info.name = strdup("nemea-module");
+   }
+   if (description != NULL) {
+      module_info.description = strdup(description);
+   } else {
+      module_info.description = strdup("");
+   }
```

Comment:
There is no check, whether any memory was allocated for module_info.name and module_info.description.